docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,339 @@
+// persona. The model use case owner a team that trains, verifies, and
+// leverages a large language model from a model frontend, drives the lifecycle
+// and rollout of new versions of those models, and defines the specific
+// performance and latency goals for the model. These model use cases are
+// expected to operate within a BackendPool sharing compute capacity with other
+// ModelUseCases, defined by the Inference Platform Admin. We allow a user who
+// has multiple UseCases across multiple pools (with the same config) to
+// specify the configuration exactly once, and deploy to many pools 
+// simultaneously. Enabling a simpler config and single source of truth
+// for a given user. ModelUseCase names are unique for a given BackendPool,
+// if the name is reused, an error will be  shown on the status of a 
+// ModelGroup that attempted to reuse. The oldest ModelUseCase, based on
+// creation timestamp, will be selected to remain valid. In the event of a race
+// condition, one will be selected at random. 
+type ModelGroup struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec ModelGroupSpec
+}
+
+type ModelGroupSpec struct {
+        // Defines the use cases in the set.
+        // UseCases can be in 2 priority classes, Critical and Noncritical. 
+        // Priority class is implicitly set to Critical by specifying an Objective.
+        // Otherwise the UseCase is considered Noncritical.
+        UseCases   []ModelUseCases
+        // Reference to the backend pools that the use cases registers to.
+        PoolRef []corev1.ObjectReference
+}
+
+// ModelGroup defines the policies for routing the traffic of a use case, this includes performance objectives 
+// and traffic splitting between different versions of the model.
+type ModelUseCases struct {
+        // The name of the model as the users set in the "model" parameter in the requests.
+        // The model name should be unique among the use cases that reference the same backend pool.
+        // This is the parameter that will be used to match the request with. In the future, we may
+        // allow to match on other request parameters. The other approach to support matching on 
+        // on other request parameters is to use a different ModelName f HTTPFilter 
+        ModelName string
+        // Optional
+        // Use cases with an objective have higher priority than use cases without.
+        // IMPORTANT: By specifying an objective, this places the UseCase in a higher priority class than UseCases without a defined priority class. 
+        // In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
+        Objective *Objective
+        // Optional.
+	// Allow multiple versions of a model for traffic splitting. 
+	// If not specified, the target model name is defaulted to the modelName parameter.
+        TargetModels []common.TargetModel
```

Comment:
Have a godoc comment describing that the model name is often a lora model.

Reply:
Done